test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsights/HttpDependencyCollectionTests.cs

```diff
@@ -0,0 +1,417 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Xunit;
+using LogLevel = Microsoft.Extensions.Logging.LogLevel;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.ApplicationInsights
+{
+    public class HttpDependencyCollectionTests
+    {
+        private const string TestArtifactPrefix = "e2etestsai";
+        private const string OutputQueueNamePattern = TestArtifactPrefix + "out%rnd%";
+        private const string TriggerQueueNamePattern = TestArtifactPrefix + "trigger%rnd%";
+        private const string InputContainerNamePattern = TestArtifactPrefix + "-in%rnd%";
+        private const string OutputContainerNamePattern = TestArtifactPrefix + "-out%rnd%";
+        private const string TriggerContainerNamePattern = TestArtifactPrefix + "-trigger%rnd%";
+        private const string _mockApplicationInsightsKey = "some_key";
+        private readonly TestTelemetryChannel _channel = new TestTelemetryChannel();
+
+        private static readonly AutoResetEvent _functionWaitHandle = new AutoResetEvent(false);
+
+        private string _inputContainerName;
+        private string _outputContainerName;
+        private string _triggerContainerName;
+        private string _outputQueueName;
+        private string _triggerQueueName;
+
+        private CloudBlobClient _blobClient;
+        private RandomNameResolver _resolver;
+        private CloudQueueClient _queueClient;
+        private CloudQueue _triggerQueue;
+
+        [Fact]
+        public async Task BindingsAreNotReportedWhenDisabled()
+        {
+            using (var host = ConfigureHost(LogLevel.Warning))
+            {
+                await host.StartAsync();
+                await host.GetJobHost()
+                    .CallAsync(typeof(HttpDependencyCollectionTests).GetMethod("BlobInputAndOutputBindings"));
```

Comment:
nit: Can you use `nameof()` for these method names? It makes navigating easier when debugging in the future (we're slowly trying to change them all as we see them).

Reply:
👍